pkg/lib/cockpit.js

```diff
@@ -3633,24 +3635,23 @@ function factory() {
                 if (watch_channel)
                     return;
 
-                const opts = {
-                    payload: "fswatch1",
-                    path,
-                    superuser: base_channel_options.superuser,
-                };
-                watch_channel = cockpit.channel(opts);
-                watch_channel.addEventListener("message", function (event, message_string) {
-                    let message;
-                    try {
-                        message = JSON.parse(message_string);
-                    } catch (e) {
-                        message = null;
-                    }
-                    if (message && message.path == path && message.tag && message.tag != watch_tag) {
-                        if (options && options.read !== undefined && !options.read)
-                            fire_watch_callbacks(null, message.tag);
-                        else
-                            read();
+                watch_channel = fsinfo(path, ["tag"], { superuser: base_channel_options.superuser });
+                watch_channel.effect(state => {
+                    if (state.error || state.info) {
+                        // Behave like fsread1, not-found is not a fatal error
+                        if (state.error && state?.error?.problem !== "not-found") {
```

Comment:
You don't need the `?.`, as you already check that state.error exists. And you have to, as you want the "else" for other problems.